ghcide/src/Development/IDE/Core/Rules.hs

```diff
@@ -1119,7 +1118,10 @@ getLinkableRule recorder =
         -- Record the linkable so we know not to unload it
         whenJust (hm_linkable =<< hmi) $ \(LM time mod _) -> do
             compiledLinkables <- getCompiledLinkables <$> getIdeGlobalAction
-            liftIO $ void $ modifyVar' compiledLinkables $ \old -> extendModuleEnv old mod time
+            liftIO $ modifyVar compiledLinkables $ \old -> do
+              let !to_keep = extendModuleEnv old mod time
```

Comment:
Surely _some_ of the explanation you put in the PR description should be written in the code for future people?

Reply:
Added.

Reply:
Sorry, forgot to include the file. I've updated it now.